src/main/webapp/meal-page-script.js

```diff
@@ -39,21 +42,79 @@ function createElementByTag(text, tag) {
   return element;
 }
 
-/** Creates a map and adds it to the page. */
+let map;
+
 function createMap() {
-  const userLocation = new google.maps.LatLng(55.746514, 37.627022);
+  let userLocation = new google.maps.LatLng(55.746514, 37.627022);
   const mapOptions = {
-    zoom: 16,
-    center: userLocation
+    zoom: 14,
+    center: userLocation,
   };
-  const map = new google.maps.Map(document.getElementById("map"), mapOptions);
-  // TODO(grenlayk): implement real geolocation here
-  // TODO(grenlayk): change "You are here" marker on custom one
+  map = new google.maps.Map(document.getElementById("map"), mapOptions);
   const locationMarker = new google.maps.Marker({
     position: userLocation,
     map,
-    title: "You are here",
+    title: "You are here (probably)",
+    label: "U",
     animation: google.maps.Animation.DROP,
   });
+
+  
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(
+      function (position) {
+        const location = new google.maps.LatLng(
+          position.coords.latitude,
+          position.coords.longitude
+        );
+        map.setCenter(location);
+        locationMarker.setPosition(location);
+        addRestaurants();
+      },
+      function () {
+        handleLocationError(true);
+        addRestaurants();
+      }
+    )
+  } else {
+    handleLocationError(false);
+    addRestaurants();
+  }
+}
+
+function addRestaurants() {
+  // Hard coded restaurants
+  const restaurants = [];
+  restaurants.push(changeLocation(map.getCenter(), 0.001, 0.001));
+  restaurants.push(changeLocation(map.getCenter(), -0.001, 0.001));
+
   // TODO(grenlayk): implement restaurants search here
-}
\ No newline at end of file
+  for (const restaurantLocation of restaurants) {
+    const marker = new google.maps.Marker({
+        position: restaurantLocation,
+        map,
+        title: "Restaurant",
+        animation: google.maps.Animation.DROP
+    });
+  }
+}
+
+function changeLocation(location, latDiff, lngDiff) {
+    const lat = location.lat() + latDiff;
+    const lng = location.lng() + lngDiff;
+    return {
+        lat, 
+        lng
+    };
+}
+
+function handleLocationError(browserHasGeolocation) {
```

Comment:
I'd suggest avoiding functions with a single, `boolean` argument that only have an `if..else` block inside.
We can have two separate functions instead.